W3ChampionsStatisticService/PersonalSettings/PersonalSettingsCommandHandler.cs

```diff
@@ -35,19 +36,35 @@ public async Task<bool> UpdatePicture(string battleTag, SetPictureCommand comman
             return true;
         }
 
-        public async Task<bool> UpdateAkaSettings(string battleTag, SetAkaSettingsCommand command)
+        public async Task UpsertSpecialPortraits(PortraitsCommand command)
         {
-            var setting = await _personalSettingsRepository.Load(battleTag);
-            if (setting == null)
+            var settings = await _personalSettingsRepository.LoadMany(command.BnetTags.ToArray());
+            
+            foreach (var playerSettings in settings)
             {
-                var playerProfile = await _playerRepository.LoadPlayerProfile(battleTag);
-                setting = new PersonalSetting(battleTag, new List<PlayerOverallStats> { playerProfile });
+                var specialPortraitsList = new List<SpecialPicture>(playerSettings.SpecialPictures);
+                foreach (var portrait in command.Portraits)
+                {
+                    specialPortraitsList.Add(new SpecialPicture(portrait, command.Tooltip));
```

Comment:
You probably want to check if portrait does not already exists. If it does, should not be added twice